manager/api/transport_test.go

```diff
@@ -0,0 +1,535 @@
+package api_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/mainflux/mainflux/manager"
+	"github.com/mainflux/mainflux/manager/api"
+	"github.com/mainflux/mainflux/manager/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+const contentType = "application/json; charset=utf-8"
+
+var (
+	validUser       = manager.User{"user@example.com", "password"}
+	invalidPassUser = manager.User{"user@example.com", "invalid_password"}
+	invalidUser     = manager.User{"userexample.com", "password"}
+	nonexistentUser = manager.User{"non-existentuser@example.com", "pass"}
+	validClient     = manager.Client{Type: "app", Name: "test_app", Payload: "test_payload"}
+	invalidClient   = manager.Client{Type: "foo", Name: "invalid_client", Payload: "some_payload"}
+	validChannel    = manager.Channel{Name: "test"}
+)
+
+func newService() manager.Service {
+	users := mocks.NewUserRepository()
+	clients := mocks.NewClientRepository()
+	channels := mocks.NewChannelRepository()
+	hasher := mocks.NewHasher()
+	idp := mocks.NewIdentityProvider()
+
+	return manager.New(users, clients, channels, hasher, idp)
+}
+
+func runService(test func(manager.Service, string, *http.Client)) {
```

Comment:
Instead of passing a test callback, rename this method to `newServer` and return its instance in the same way it is done at `newService`. You can obtain all of the data in the test itself.

Just to be clear, this is not a bad way to remove duplication, but I'd prefer that we are as explicit as possible in our test cases.